firestarter/workflows/build_images/build_images.py

```diff
@@ -160,16 +160,12 @@ def execute(self):
         self.filter_on_premises()
 
         print(f"Building '{self.repo_name}' from '{self.from_point}' for '{self.on_premises}'")
-
-        if self.login_required:
-            # Log in to the Azure Container Registry for each on-premises active in the configuration file
-            for key in self.on_premises:
-                # Log in to the Azure Container Registry
-                registry = self.config.images[key].registry
-                cli = get_default_cli()
-                success = cli.invoke(['acr', 'login', '--name', registry])
-                if success != 0:
-                    raise Exception('Login to the Azure Container Registry failed.')
+        
+        # Log in to the Azure Container Registry for each on-premises active in the configuration file
+        for key in self.on_premises:
+            print(self.config.images[key])
+            provider = DockerRegistryAuthFactory.provider_from_str(self.config.images[key].auth_strategy)
+            provider.login_registry(self.config.images[key].registry, provider.get_registry_auth())
```

Comment:
I would set the registry name as a property of the "Provider".

I would also put the `get_registry_auth()` as a required method to be defined  by the concrete provider, and called by the login_registry() method, something like:
```
provider.login_registry()
```

```python
def login_registry(self):
   self.get_registry_auth()
   ...
```